File: medium/reverse_int.cpp
class Solution {
public:
    int reverse(int x) {
        if(x == 0){
            return 0;
        }
        
        long long int_max = 2147483647;
        long long int_min = -2147483648;


        long long y = (long long)x;

        bool negative = y < 0 ? true : false;

        y = abs(y);

        long long out = 0;
        while (y != 0){
            out = (y % 10) + out * 10;
            y /= 10;
        }

        if (negative){
            out *= -1;
        }

        if(out < int_min || out > int_max){
            return 0;
        }

        return (int) out;


    }
};

File: medium/reverse_int.py
class Solution(object):
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """

        if x == 0:
            return x

        int_max = 2147483647
        int_min = -2147483648

        negative = True if x < 0 else False

        x = abs(x)
        out = 0
        while x != 0:
            out = (x % 10) + out * 10
            x //= 10


        if negative:
            out *= -1

        if out < int_min or out > int_max:
            return 0
        
        return out
